merlin/table/cupy_column.py

```diff
@@ -93,7 +93,9 @@ def _register_to_dlpack_from_cupy():
     import cupy as cp
 
     @_to_dlpack.register(cp.ndarray)
-    def _to_dlpack_from_tf_tensor(tensor):
+    def _to_dlpack_from_cp_tensor(tensor):
+        if tensor.dtype == cp.dtype("bool"):
+            tensor = tensor.astype(cp.dtype("int8"))
```

Comment:
Do we need a corresponding coercion back to bool dtypes on the other side (the from dlpack functions?)

Reply:
No we dont, we have seen in testing that the models seem to be ok with the conversion to 1 and 0 and they are able to interpret those values correctly. We were originally going to make that  change also (it belongs in the tensor column constructor) but during testing we saw that is was not necessary.

Reply:
I'm not 100% sure these values are being interpreted as bools on the other side (vs being treated as `int8` and having that still function for model training.) I'm not sure if it matters though, and I'd rather avoid an extra conversion if we can for performance reasons. So...I'm on the fence about adding the return conversion. "Let's see if it turns out to be a problem not to have it" seems reasonable, I suppose.